src/main/scala/scalismo/kernels/DiscreteMatrixValuedPDKernel.scala

```diff
@@ -25,13 +25,13 @@ import scalismo.geometry.{ NDSpace, Dim, SquareMatrix }
  *  Mathematically, it can be represented as a covariance matrix. However, it has more structure, i.e. its entry ij
  *  is a matrix. Furthermore, the class has the knowledge about its domain (the point on which it is defined).
  */
-class DiscreteMatrixValuedPDKernel[D <: Dim: NDSpace, DO <: Dim: NDSpace] private[scalismo] (val domain: DiscreteDomain[D],
-    val k: (PointId, PointId) => SquareMatrix[DO]) {
+class DiscreteMatrixValuedPDKernel[D <: Dim: NDSpace] private[scalismo] (
+    val domain: DiscreteDomain[D],
+    val k: (PointId, PointId) => DenseMatrix[Double],
+    val outputDim: Int) {
```

Comment:
By having the outputdim as a field, we lose the type safety where we could check that the output dimentionalities matched. Do you think it would make sense to also rely on an implicit vectorizer here (even if we don't use it), to figure out the output dimensionality from the vectorizer. The compiler might have more chances to help out in case of mixing kernels with different output dimensionality.